### DISCUSSION TITLE
merge several sweeps to use with "pyart.map.grid_from_radars"

### DISCUSSION THREAD

**Discussion by johnnypeste** [ARM-DOE/pyart#1693] [General]
I have 3 sigmet/iris/vaisala radar files, each one with a different elevation, namely:

FAR241015000002.RAWMZAD (hour 00 minute 00 second 02) is the lowest elevation
FAR241015000023.RAWML13  (hour 00 minute 00 second 23) is the next elevation
FAR241015000044.RAWML14 ((hour 00 minute 00 second 44) is the next elevation

I could process each of the 3 filenames in separate with:

radar = pyart.io.read_sigmet(filename)
grid = pyart.map.grid_from_radars(radar,
                                  grid_shape=(z_grid_points,
                                              y_grid_points,
                                              x_grid_points),
                                  grid_limits=(z_grid_limits,
                                               y_grid_limits,
                                               x_grid_limits),
                                 )

but I think it would be a better interpolation to assume that the timestamp is the same for those 3 files and merge them into a single file to feed in "pyart.map.grid_from_radars".

Am I right and what is the best way to achieve this?

**Comment by zssherman** [comment]
@johnnypeste I think joining them together if they are by elevation or sweeps should be fine with the pyart.utils.join_radar([sweep0, sweeps2, sweep3]) etc. The gridding does a ROI of a gate and around it within meters so you'll be grabbing data in x, y and z. Seeing as the timestamps are within seconds im assuming the radar for that volume is just split for separate files by sweep. You could also possible join sweeps together using xradar https://docs.openradarscience.org/projects/xradar/en/stable/datamodel.html and use the pyart functionality to go from a dataset to a radar object.

**Reply by johnnypeste** [reply]
import xradar as xd
import pyart

tree0 = xd.io.open_iris_datatree("FAR241015000002.RAWMZAD")
tree1 = xd.io.open_iris_datatree("FAR241015000023.RAWML13" )
tree2 = xd.io.open_iris_datatree("FAR241015000044.RAWML14")

Now...

How do I use xradar to merge tree0  tree1 and  tree2 into a single dataset "tree" with the 3 elevations??

Later I would do:

radar = tree.pyart.to_radar()

and could use: 

 grid = pyart.map.grid_from_radars((radar,1),  ... etc

**Reply by johnnypeste** [reply]
I mean:

grid = pyart.map.grid_from_radars((radar,), ... etc
